src/fprime_gds/common/loaders/json_loader.py

```diff
@@ -0,0 +1,218 @@
+        Caches the constructed EnumType object in the parsed_types dictionary.
+
+        Args:
+            type_name (str): The name of the enum type.
+            qualified_type (dict): A dictionary containing the qualified type information.
+
+        Returns:
+            EnumType: The constructed EnumType object.
+
+        """
+        enum_dict = {}
+        for member in qualified_type.get("enumeratedConstants"):
+            enum_dict[member.get("name")] = member.get("value")
+        enum_type = EnumType.construct_type(
+            type_name,
+            enum_dict,
+            qualified_type.get("representationType").get("name"),
+        )
+        self.parsed_types[type_name] = enum_type
+        return enum_type
+
+    def construct_array_type(self, type_name: str, qualified_type: dict) -> ArrayType:
+        """
+        Constructs an ArrayType object based on the given type name and qualified type dictionary.
+        Caches the constructed ArrayType object in the parsed_types dictionary.
+
+        Args:
+            type_name (str): The name of the array type.
+            qualified_type (dict): The qualified type dictionary containing information about the array type.
+
+        Returns:
+            ArrayType: The constructed ArrayType object.
+
+        """
+        array_type = ArrayType.construct_type(
+            type_name,
+            self.parse_type(qualified_type.get("elementType")),
+            qualified_type.get("size"),
+            qualified_type.get("elementType").get("format", "{}"),
+        )
+        self.parsed_types[type_name] = array_type
+        return array_type
+
+    def construct_serializable_type(
+        self, type_name: str, qualified_type: dict
+    ) -> SerializableType:
+        """
+        Constructs a SerializableType based on the given type name and qualified type dictionary.
+        Caches the constructed SerializableType object in the parsed_types dictionary.
+
+        Args:
+            type_name (str): The name of the serializable type.
+            qualified_type (dict): The qualified type dictionary containing information about the type.
+
+        Returns:
+            SerializableType: The constructed serializable type.
+
+        """
+        struct_members = []
+        for name, member_dict in qualified_type.get("members").items():
+            member_type_dict = member_dict.get("type")
+            member_type_obj = self.parse_type(member_type_dict)
+
+            # For member arrays (declared inline, so we create a type on the fly)
+            if member_dict.get("size") is not None:
+                member_type_obj = ArrayType.construct_type(
+                    f"Array_{member_type_obj.__name__}_{member_dict.get('size')}",
+                    member_type_obj,
+                    member_dict.get("size"),
+                    member_dict.get("type").get("format", "{}"),
+                )
+            # TODO: does this need to be preprocessed or something?
```

Comment:
Probably?  Seems like you have a FPP format string parser function, so it should be called on it, yes?